### PR TITLE
Add article about connecting to a Hotel's captive portal

### PR DESCRIPTION
**Pull request by paultcochrane**
The article describes the process of debugging a strange hotel hotspot wireless
connection and the Perl-based solution using WWW::Mechanize.

Originally appeared here: https://peateasea.de/hotel-hotspot-hijinks/.

@oalders spotted this on Mastodon and recommended that I submit the article as a PR.  Here 'tis ;-)

### CODE REVIEW

File: content/article/hotel-hotspot-hijinks.md [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+$mech->get('http://google.com');
+
+# check that we have the login page and not Google or something else
+# i.e. we're not logged in
+if ($mech->title() =~ 'login') {
+    $mech->submit_form(
+        form_name => 'login',
+        fields => {
+            username => 'whg4@haus-reimann',
+            password => 'HRAltenau4'
```
Comment (oalders) [inline content/article/hotel-hotspot-hijinks.md:268]: First off, thanks very much! Someone pointed out that this is probably enough information to deduce the name of the hotel (and possibly the room in question). I think obscuring that would not make the article any less useful?

Comment (paultcochrane) [reply content/article/hotel-hotspot-hijinks.md:268]: Bother!  I obviously forgot to change that after copy-pasting the "production" code into the article. Damn!  Thanks heaps for pointing that out!

Comment (paultcochrane) [reply content/article/hotel-hotspot-hijinks.md]: Ok, I've fixed the offending lines.  I feel really daft now for having added that.  It certainly wasn't my intention :-/

Comment (oalders) [reply content/article/hotel-hotspot-hijinks.md]: I once left hard-coded credentials in a script that was included in a blog post. A very kind soul emailed me to point that out before any harm was done.

Comment (paultcochrane) [reply content/article/hotel-hotspot-hijinks.md]: Thanks for helping me out on that one; I appreciate it! :-)

File: content/article/hotel-hotspot-hijinks.md [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+## The conclusion
+
+Well, this script certainly saved me some time and hassle when waking up my
+laptop from the suspend state.  Also, it was fun working out what pieces of
+the puzzle were necessary in order to build a solution.  Perl saves the
+day[^perl-saves-the-day] again!
+
+[^perl-saves-the-day]: Well, it saved the week really.
+
+Originally posted on [https://peateasea.de](https://peateasea.de/hotel-hotspot-hijinks/).
```
Comment (oalders) [inline content/article/hotel-hotspot-hijinks.md:300]: I'm glad you added this link. Hopefully it can drive some traffic over to your site.

**Review by oalders — APPROVED** [review]
Wonderful stuff. Thank you so much for sharing your work!

**Pull request merged**